src/trading/CompleteSets.sol

```diff
@@ -48,11 +49,12 @@ contract CompleteSets is Controlled, ReentrancyGuard, ICompleteSets {
         uint8 _numOutcomes = _market.getNumberOfOutcomes();
         ERC20 _denominationToken = _market.getDenominationToken();
         uint256 _marketCreatorFeeRate = _market.getMarketCreatorSettlementFeeInAttoethPerEth();
-        uint256 _marketCreatorFee = _amount.mul(_marketCreatorFeeRate).div(1 ether);
+        uint256 _payout = _amount.mul(_market.getMarketDenominator());
```

Comment:
If we can spare the local variable, we should read `_market.getMarketDenominator()` once and reuse it.  Reading from storage is ~200 gas and calling a cross-contract function is 700 gas or something.